includes/polling/loadbalancers/f5-ltm.inc.php

```diff
@@ -0,0 +1,193 @@
+ * the source code distribution for details.
+ */
+
+// Define some error messages
+$error_poolaction = array();
+$error_poolaction[0] = "Unused";
+$error_poolaction[1] = "Reboot";
+$error_poolaction[2] = "Restart";
+$error_poolaction[3] = "Failover";
+$error_poolaction[4] = "Failover and Restart";
+$error_poolaction[5] = "Go Active";
+$error_poolaction[6] = "None";
+
+$component = new LibreNMS\Component();
+$options['filter']['disabled'] = array('=',0);
+$options['filter']['ignore'] = array('=',0);
+$components = $component->getComponents($device['device_id'], $options);
+
+// We only care about our device id.
+$components = $components[$device['device_id']];
+
+// We extracted all the components for this device, now lets only get the LTM ones.
+$keep = array();
+$types = array('f5-ltm-vs', 'f5-ltm-pool', 'f5-ltm-poolmember');
+foreach ($components as $k => $v) {
+    foreach ($types as $type) {
+        if ($v['type'] == $type) {
+            $keep[$k] = $v;
+        }
+    }
+}
+$components = $keep;
+
+// Only collect SNMP data if we have enabled components
+if (count($components > 0)) {
+    // Let's gather the stats..
+    $ltmVirtualServStatEntry = snmpwalk_array_num($device, '.1.3.6.1.4.1.3375.2.2.10.2.3.1', 0);
+    $ltmPoolMemberStatEntry = snmpwalk_array_num($device, '.1.3.6.1.4.1.3375.2.2.5.4.3.1', 0);
+    $sysGlobalHttpStat = snmpwalk_array_num($device, '.1.3.6.1.4.1.3375.2.1.1.2.4', 0);
+
+    // and check the status
+    $ltmVsStatusEntry = snmpwalk_array_num($device, '1.3.6.1.4.1.3375.2.2.10.13.2.1', 0);
+    $ltmPoolMbrStatusEntry = snmpwalk_array_num($device, '1.3.6.1.4.1.3375.2.2.5.6.2.1', 0);
+    $ltmPoolEntry = snmpwalk_array_num($device, '1.3.6.1.4.1.3375.2.2.5.1.2.1', 0);
+
+    // Loop through the components and extract the data.
+    foreach ($components as $key => &$array) {
+        $type = $array['type'];
+        $UID = $array['UID'];
+        $label = $array['label'];
+        $hash = $array['hash'];
+        $rrd_name = array($type, $label, $hash);
+
+        if ($type == 'f5-ltm-vs') {
+            $rrd_def = array(
+                'DS:pktsin:COUNTER:600:0:U',
+                'DS:pktsout:COUNTER:600:0:U',
+                'DS:bytesin:COUNTER:600:0:U',
+                'DS:bytesout:COUNTER:600:0:U',
+                'DS:totconns:COUNTER:600:0:U',
+            );
+
+            $fields = array(
+                'pktsin' => $ltmVirtualServStatEntry['1.3.6.1.4.1.3375.2.2.10.2.3.1.6.'.$UID],
+                'pktsout' => $ltmVirtualServStatEntry['1.3.6.1.4.1.3375.2.2.10.2.3.1.8.'.$UID],
+                'bytesin' => $ltmVirtualServStatEntry['1.3.6.1.4.1.3375.2.2.10.2.3.1.7.'.$UID],
+                'bytesout' => $ltmVirtualServStatEntry['1.3.6.1.4.1.3375.2.2.10.2.3.1.9.'.$UID],
+                'totconns' => $ltmVirtualServStatEntry['1.3.6.1.4.1.3375.2.2.10.2.3.1.11.'.$UID],
+            );
+
+            // Let's print some debugging info.
+            d_echo("\n\nComponent: ".$key."\n");
```

Comment:
I have removed all the counter data as the RRD debugs already show these, Not yet pushed.